test/src/code_field/code_controller_indent_selection_test.dart

```diff
@@ -0,0 +1,477 @@
+        );
+        expect(
+          controller.code.visibleText,
+          controller.text,
+          reason: example.name,
+        );
+        expect(
+          controller.value.selection,
+          example.expectedSelection,
+          reason: example.name,
+        );
+      }
+    });
+
+    test('Selection is a range', () {
+      final examples = [
+        _Example(
+          name: 'WHEN non-collapsed selection, two lines, not first, not last. '
+              'SHOULD add indentation to the selected lines',
+          initialFullText: '''
+aaaa
+aaAA
+AAAa
+aaaa
+''',
+          expectedFullText: '''
+aaaa
+${indent}aaAA
+${indent}AAAa
+aaaa
+''',
+          expectedVisibleText: '''
+aaaa
+${indent}aaAA
+${indent}AAAa
+aaaa
+''',
+          initialSelection: const TextSelection(
+            baseOffset: 7,
+            extentOffset: 13,
+          ),
+          expectedSelection: TextSelection(
+            baseOffset: 7 + indentLength,
+            extentOffset: 13 + indentLength * 2,
+          ),
+        ),
+        _Example(
+          name: 'WHEN entire document is selected '
+              'SHOULD add indentation to all lines',
+          initialFullText: '''
+AAA
+AAA
+AAA
+''',
+          expectedFullText: '''
+${indent}AAA
+${indent}AAA
+${indent}AAA
+''',
+          expectedVisibleText: '''
+${indent}AAA
+${indent}AAA
+${indent}AAA
+''',
+          initialSelection: const TextSelection(
+            baseOffset: 0,
+            extentOffset: 11,
+          ),
+          expectedSelection: TextSelection(
+            baseOffset: 0 + indentLength,
+            extentOffset: 11 + indentLength * 3,
+          ),
+        ),
+      ];
+
+      for (final example in examples) {
+        controller.text = example.initialFullText;
+        controller.selection = example.initialSelection;
+        controller.indentSelection();
+
+        expect(
+          controller.value.text,
+          example.expectedVisibleText,
+          reason: example.name,
+        );
+        expect(
+          controller.code.text,
+          example.expectedFullText,
+          reason: example.name,
+        );
+        expect(
+          controller.code.visibleText,
+          controller.text,
+          reason: example.name,
+        );
+        expect(
+          controller.value.selection,
+          example.expectedSelection,
+          reason: example.name,
+        );
+      }
+    });
+  });
+
+  group('CodeController.indentSelection() => Folded text', () {
+    final language = java;
+    CodeController controller = CodeController(
+      params: const EditorParams(tabSpaces: 2),
+      language: language,
+    );
+    final indentLength = controller.params.tabSpaces;
+    final indent = ' ' * indentLength;
+
+    test('description', () {
+      final examples = [
+        _Example(
+          name: 'Indentation through a folded block affects and unfolds it',
+          initialFullText: '''
+class MyClass {
+  void getSmth(){
```

Comment:
This method name is non-neutral, i.e. the reader may think it is important for the test. I like `method1`, `method2`, etc. for such tests.